pyicloud/services/drive.py

```diff
@@ -0,0 +1,150 @@
+from datetime import datetime, timedelta
+import json
+import re
+import sys
+
+
+class DriveService(object):
+
+    def __init__(self, service_root, session, params):
+        self._service_root = service_root
+        self.session = session
+        self.params = params
+        self._root = None
+
+    def get_node_data(self, id):
+        request = self.session.post(
+            self._service_root + '/retrieveItemDetailsInFolders',
+            params=self.params,
+            data=json.dumps([{
+                "drivewsid": "FOLDER::com.apple.CloudDocs::%s" % id,
+                "partialData": False,
+            }])
+        )
+        return request.json()[0]
+
+    def get_token_from_cookie(self):
+        for cookie in self.session.cookies:
+            if cookie.name == 'X-APPLE-WEBAUTH-TOKEN':
+                match = re.search(r'\bt=([^:]+)', cookie.value)
+                if not match:
+                    raise Exception("Can't extract token from %r" % (
+                        token_cookie,
+                    ))
+                return match.group(1)
+        raise Exception("Token cookie not found")
+
+    def get_file(self, id, **kwargs):
+        meta = self.session.get(
+            self._service_root + '/ws/com.apple.CloudDocs/download/by_id',
+            params={
+                'document_id': id,
+                'token': self.get_token_from_cookie(),
```

Comment:
The token is set on auth right? So you should be able to store it in __init__?

Reply:
That's a good point I'll make that change.

Reply:
Done